app/models/piece.rb

```diff
@@ -93,4 +93,20 @@ def remains_on_board?(to_x:, to_y:)
       false
     end
   end
+
+  def successful_capture(to_x:, to_y:)
+    return false if obstructed_diagonally?(to_x:, to_y:)
+    return false if obstructed_vertically?(to_x:, to_y:)
+    return false if obstructed_horizontally?(to_x:, to_y:)
+    current_color = current_piece.color
+    target_piece = Piece.where(x_position: to_x, y_position: to_y, active: true, color: !current_color)  # Add requirement that target MUST be opponent's color
+    if target_piece.exists?
+      return false if target_piece == king  # King can't be captured, only checkmated
```

Comment:
This works, but would be nice to raise some kind of error so the game knows the capture failed because we were trying to attack a king. That could be added later I suppose.